core/actionProxy/actionproxy.py

```diff
@@ -46,59 +47,50 @@ def __init__(self, source = None, binary = None):
     # @return True iff binary exists and is executable
     def init(self, message):
         def prep():
-            if 'code' in message:
+            if 'code' in message and message['code'] is not None:
                 binary = message['binary'] if 'binary' in message else False
                 if not binary:
-                    with codecs.open(self.source, 'w', 'utf-8') as fp:
```

Comment:
I struggle with git...  Sorry.

I would have rewritten this as `if message.get('code'):` which fails if 1) 'code' not in message, 2) message['code'] in [None, '', False, 0, 0.0, [], {}, set()]